### ISSUE TITLE
Marginal Tax Rate computation bug

### ISSUE THREAD

**Issue by PavelMakarchuk** [PolicyEngine/policyengine-us#3071]
In the `[following scenario](https://policyengine.org/us/household?focus=householdOutput.mtr&household=34364)` the MTR does not seem to be computed. 

In [some instances](https://policyengine.org/us/household?focus=householdOutput.mtr&reform=2&region=co&timePeriod=2023&baseline=29166&household=34369) it seems to be not computed in combination with a reform.

It seems to occur sporadically as the same household compositions have sometimes returned and sometimes not returned MTRs.

@nikhilwoodruff @anth-volk

**Comment by anth-volk** [comment]
@nikhilwoodruff Just want to confirm that this doesn't appear to be driven by my recent changes on [API #230](https://github.com/PolicyEngine/policyengine-api/issues/230)?

Unfortunately, I probably wouldn't have time on this this week, perhaps also not next, but I'll try to find some time.

**Comment by PavelMakarchuk** [comment]
It seems to come up every time I reload a saved household

**Comment by anth-volk** [comment]
Could this have anything to do with PR #3021? I realize it wasn't fully committed, but thought there might be some overlap

**Comment by martinholmer** [comment]
@anth-volk said in issue #3071:
> Could this have anything to do with [issue] #3021 [that generated PR #3036]? 
> I realize it wasn't **fully committed**, but thought there might be some overlap.

I think it is plausible to think there is "some overlap."
However, PR #3036 has not even been reviewed much less committed.

**Comment by MaxGhenis** [comment]
I just hit this bug, could it also be related?

<img width="839" alt="image" src="https://github.com/PolicyEngine/policyengine-api/assets/6076111/5d4a230f-6900-4902-850f-63a2ee266ae2">

https://policyengine.org/us/household?region=co&timePeriod=2023&baseline=2&focus=householdOutput.earnings&reform=30092&household=34494

**Comment by martinholmer** [comment]
@MaxGhenis said:
> I just hit this bug [illustrated by a webapp screenshot], could it also be related?

Could you explain what is unexpected for those who are not familiar with the webapp?

**Comment by martinholmer** [comment]
@MaxGhenis,  So you mean in issue #3071 that the `household_net_income` on the Y-axis is **tiny** in comparison to the level of `employment_income` for the household head on the X-axis?

I have no idea what is going on in the webapp, but the two related issues I have submitted --- issue #3017 and issue #3029 --- show clearly that there are problems with the calculation of marginal tax rates and cliff gaps in the PolicyEngine-US source code.

Why don't you answer your own question by creating a new PolicyEngine-US test (using the case in your webapp example) to see if the test gives the expected values for `household_net_income`.   If so, then the problem is probably in the webapp.

**Comment by anth-volk** [comment]
Just did some testing - I think this issue must have something to do with the PE-US package and not the web app. The key is that, for some reason, household #34494, which has no household income, receives a household_net_income of 0 when simulated under US default policy. One I just created, #34553, also has no net income, but when placed under US default policy, it's simulated to have a net income of around $20,500. 

If you notice, too, #34494 has no default "Net income" option in the variable dropdown menu, perhaps because the output 0 is falsy? I'm sure this then also ruins the calculation of "Market income."

I even tested it all by manually inputting all the household variable definitions that #34494 has - single, 1 child, Colorado, no income for anyone, ages 40 & 5, and received a working output, though I didn't add any reform policy, as it didn't seem like that was included in #34494.

So my thoughts are something like this. I think it's possible that either there was a change to Colorado calculation specifically that somehow breaks this (though that seems unlikely), or there is some particular variable that was input when #34494 was created that broke this (would be helpful to know if some alternative value was input). Otherwise I'm at a bit of a loss, but it doesn't appear to be a front-end or API thing.

**Comment by martinholmer** [comment]
@anth-volk said:
> Just did some testing - I think this issue [#3071] must have something to do with the PE-US package and not the web app. 

Can you share with us the YAML test file(s) you wrote to test PolicyEngine-US?

**Comment by MaxGhenis** [comment]
@martinholmer I don't think this is related to the issues/PRs you've filed since they only occur in rare occasions in the web app, and in the cases I've looked at the results from the Python package seem fine. Your changes seem like they would affect a much wider range of households (all except California right?).

**Comment by anth-volk** [comment]
@martinholmer To be clear, I didn’t write any unit tests, only used browser debugging and manual inspection of API returns
